cli/src/commands/file/track.rs

```diff
@@ -44,26 +52,87 @@ pub(crate) fn cmd_file_track(
     command: &CommandHelper,
     args: &FileTrackArgs,
 ) -> Result<(), CommandError> {
-    let mut workspace_command = command.workspace_helper(ui)?;
+    let (mut workspace_command, auto_stats) = command.workspace_helper_with_stats(ui)?;
     let matcher = workspace_command
         .parse_file_patterns(ui, &args.paths)?
         .to_matcher();
+
     let options = workspace_command.snapshot_options_with_start_tracking_matcher(&matcher)?;
 
     let mut tx = workspace_command.start_transaction().into_inner();
     let (mut locked_ws, _wc_commit) = workspace_command.start_working_copy_mutation()?;
-    let (_tree_id, stats) = locked_ws.locked_wc().snapshot(&options)?;
+    let (_tree_id, track_stats) = locked_ws.locked_wc().snapshot(&options)?;
     let num_rebased = tx.repo_mut().rebase_descendants()?;
     if num_rebased > 0 {
         writeln!(ui.status(), "Rebased {num_rebased} descendant commits")?;
     }
     let repo = tx.commit("track paths")?;
     locked_ws.finish(repo.op_id().clone())?;
-    print_snapshot_stats(
+    print_track_snapshot_stats(
         ui,
-        &stats,
+        auto_stats,
+        track_stats,
+        &matcher,
         workspace_command.env().path_converter(),
-        SnapshotContext::Track,
     )?;
     Ok(())
 }
+
+pub fn print_track_snapshot_stats(
+    ui: &Ui,
+    auto_stats: SnapshotStats,
+    track_stats: SnapshotStats,
+    track_matcher: &dyn Matcher,
+    path_converter: &RepoPathUiConverter,
+) -> io::Result<()> {
+    if track_stats
+        .untracked_paths
+        .iter()
+        .any(|(path, _)| track_matcher.matches(path))
+        .not()
+    {
+        // we can use the "normal" report path, nothing special happened to a file we
+        // care about
+        return print_snapshot_stats(ui, &auto_stats, path_converter);
+    }
```

Comment:
Remove this? I don't think we need to use the "normal" message here because the user intentionally tried to start tracking some new files.